### ISSUE TITLE
Optional coverage feature creates too many NoMethodErrors

### ISSUE THREAD

**Issue by drbrain** [lanej/cistern#49]
While debugging an application using cistern I needed to run with `-d` to determine the cause of a segmentation fault (it was a SystemStackError that overflowed just a bit too much).

However, when looking at the output, I see many repetitions of:

```
Exception `NoMethodError' at /…/gems/cistern-2.2.3/lib/cistern/attributes.rb:126 - undefined method `+' for nil:NilClass
```

This is [line 126 of attributes.rb](https://github.com/lanej/cistern/blob/v2.2.3/lib/cistern/attributes.rb#L126) and it should use feature detection like [the accompanying work over here](https://github.com/lanej/cistern/blob/v2.2.3/lib/cistern/attributes.rb#L63-L72).  Without this a stack trace is generated and destroyed every time an attribute is read (which can be a large object).

**Comment by lanej** [comment]
oh man, inline `rescue` is my least favorite ruby feature. will remove.

**Comment by lanej** [comment]
Released in v2.2.4
